drivers/hmis_external_apis/lib/ac_hmis/import_ce_match_rules_20251001.rb

```diff
@@ -0,0 +1,211 @@
+    def import!
+      puts "Starting CE Rules import from #{@csv_file_path}"
+      puts '=' * 50
+
+      Hmis::Hud::Base.transaction do
+        puts 'Initializing rules...'
+        rules = []
+        CSV.foreach(@csv_file_path, headers: true) do |row|
+          rule_data = process_row(row)
+          rules << rule_data if rule_data
+        end
+
+        rules_to_create = rules.select(&:new_record?)
+        rules_to_keep = rules.select(&:persisted?)
+        puts "Found #{rules_to_create.size} rules to import, and #{rules_to_keep.size} rules to keep..."
+
+        to_delete = Hmis::Ce::Match::Rule.eligibility_requirement.
+          where(owner: Hmis::Hud::Project.hmis.where(id: @project_ids)).
+          where.not(id: rules_to_keep.map(&:id))
+        puts "Deleting #{to_delete.count} existing Project-level eligibility rules for this set of projects that we\'re not keeping..."
+        to_delete.delete_all
+
+        to_delete = Hmis::Ce::Match::Rule.eligibility_requirement.
+          where(owner: Hmis::UnitGroup.where(project_id: @project_ids)).
+          where.not(id: rules_to_keep.map(&:id))
+        puts "Deleting #{to_delete.count} existing UnitGroup-level eligibility rules for this set of projects that we\'re not keeping..."
+        to_delete.delete_all
+
+        puts 'Importing...'
+        Hmis::Ce::Match::Rule.import!(rules_to_create)
+
+        puts 'Setting up Unit Group rules...'
+        unit_group_default_rules_to_create = build_default_unit_group_rules
+        puts "Importing #{unit_group_default_rules_to_create.size} default unit group rules..."
+        Hmis::Ce::Match::Rule.import!(unit_group_default_rules_to_create)
+      end
+
+      print_summary
+    end
+
+    private
+
+    # Processes a single CSV row.
+    # Returns Hmis::Ce::Match::Rule object or nil if error.
+    # The returned rule object may be new or existing.
+    def process_row(row)
+      project_id = row['Project ID']&.strip
+      rule_name = row['Rule Name']&.strip
+      rule_expression = row['Rule Expression']&.strip
+
+      # Validate required fields
+      if project_id.blank? || rule_name.blank? || rule_expression.blank?
+        error_msg = "Missing required fields for row: Project ID=#{project_id}, Rule Name=#{rule_name}, Rule Expression=#{rule_expression}"
+        @errors << error_msg
+        puts "ERROR: #{error_msg}"
+        return
+      end
+
+      # Find the project
+      project = Hmis::Hud::Project.hmis.find_by(ProjectID: project_id)
+      unless project
+        error_msg = "Project with ID #{project_id} not found"
+        @errors << error_msg
+        puts "ERROR: #{error_msg}"
+        return
+      end
+      @project_ids << project.id
+
+      # Skip rows where Unit Group Name is not null
+      unit_group = nil
+      if row['Unit Group Name'].present?
+        group_name = row['Unit Group Name']
+        unit_group = project.unit_groups.find_by(name: group_name)
```

Comment:
could we have dupes accidentally?
```suggestion
        unit_group = project.unit_groups.where(name: group_name).sole
```